catalog/spatial/registry/registry-source-configuration-handler/src/main/java/org/codice/ddf/registry/source/configuration/SourceConfigurationHandler.java

```diff
@@ -236,10 +236,11 @@ private synchronized void updateRegistryConfigurations(Metacard metacard, boolea
             }
             Configuration curConfig = findOrCreateConfig(factoryPid,
                     fpidToConfigurationMap,
+                    factoryPidMask,
                     (autoActivateConfigurations && factoryPidMask.equals(bindingTypeToActivate)));
 
             Hashtable<String, Object> serviceConfigurationProperties = new Hashtable<>();
-            if (fpidToConfigurationMap.containsKey(curConfig.getFactoryPid())) {
+            if (fpidToConfigurationMap.containsKey(curConfig.getFactoryPid() + curConfig.getProperties().get(BINDING_TYPE))) {
```

Comment:
There will not always be a bindingType property. In the case where there isn't a bindingType, the String "null" will be appended to the factoryPid. These values aren't outward facing, just used as the key for the fpidToConfigurationMap.